### DISCUSSION TITLE
How do you convert the image preview in finder for a grayscale image that was converted from a color image

### DISCUSSION THREAD

**Discussion by djoins** [ImageMagick/ImageMagick#3451] [Digital Image Processing] [answered]
I have a script to convert images into grayscale tiffs. All is working well except, when the file is complete, it is still showing a color image in mac's finder preview thumbnail. I am not sure if there is an option I am overlooking. I have played around with -preview type and it is updating the thumbnail preview but is creating a contact sheet. Not sure if I am going about it the right way with -preview or what actually controls this. Any help would be much appreciated.

```
for f in "$@"
do
    echo "$f"
    /usr/local/bin/convert -density 300 "$f"[0] -resize 1000x1000 -density 300 -type grayscale \
      -define tiff:endian=msb -compress LZW -format tif "${f%.*}_GS.tiff" && mv "${f%.*}_GS.tiff" /Done && rm "$f"

done
```

**Comment by snibgo** [comment]
I know nothing about "mac's finder preview thumbnail".

What does exiftool say about the converted tiffs? I expect exiftool says there is one sample per pixel. If so, then your IM command has worked fine, creating a grayscale image.

**Reply by djoins** [reply]
It is creating as a grayscale image, when opened in photoshop or other applications all is correct and is shown as grayscale. It is just more of an annoyance when users are searching for images visually, the thumbnail displayed in finder is still in color. I am attaching screen shot of image thumbnail from finder and the same image opened in photoshop. But all in all the end result is working fine. 
![Screen Shot 2021-03-26 at 12 54 48 PM](https://user-images.githubusercontent.com/81434659/112673464-be785c00-8e32-11eb-8c96-566d3043a907.png)
![Screen Shot 2021-03-26 at 12 55 23 PM](https://user-images.githubusercontent.com/81434659/112673465-be785c00-8e32-11eb-9584-7353565549ae.png)

**Comment by fmw42** [comment]
I have had this issue also. Are previews regenerated when processing the main image?

**Reply by fmw42** [reply]
Post an example input TIFF and the exact command line that you use to process it (not in a loop or script) so that we have actual arguments. Also identify your IM version and platform.

**Comment by snibgo** [comment, marked as answer]
I suppose the thumbnail is stored in your input file, as metadata, which IM simply copies from input to output. IM "-strip" might remove it. Or use exiftool to remove it.

**Reply by djoins** [reply]
Thats it! "-strip" worked perfectly. Thank you!

**Reply by fmw42** [reply]
I suppose it gets removed and then regenerated from the modified image.
